Allow configuring retries and timeout.

serhalp opened the pull request:
- timeout was not enabled nor configurable previously
- retries were hardcoded to `3` and not configurable

Fixes https://github.com/segmentio/analytics-node/issues/119.

I looked into writing tests, but you've got the [mock server set up in a `before`](https://github.com/segmentio/analytics-node/blob/4290503572a7c9661661dbcafef75b573cdbda0b/test.js#L34-L49) so I would need to refactor the tests to allow a test or two to configure the server to conditionally fail with a retryable error or not respond. If the lack of tests is a blocker, I can give it a try.

Thanks!

stephenmathieson commented on index.js:
```diff
@@ -49,6 +49,8 @@ class Analytics {
     this.queue = []
     this.writeKey = writeKey
     this.host = removeSlash(options.host || 'https://api.segment.io')
+    this.retries = options.retries || 3
```
"retries" sounds like boolean to me. Maybe something a bit more descriptive like "retryCount" would be better?

/cc @stevenmiller888

nettofarah replied:
`retryCount` sounds a little less ambiguous to me too.
I'd be down with that.

nettofarah commented:
Great job, @serhalp.
I still think tests would be pretty critical here before merging.
Do you think you could write at least one test to take care of this scenario.

I was thinking we could add a second condition to the mock server.
something like:

```javascript
if (batch[0] === 'timeout') {
  return setTimeout(() => res.end(), 5000)  
}
```

serhalp commented:
@nettofarah @stephenmathieson thanks, I renamed `retries` to `retryCount` and added a test for the timeout. Hope you're ok with adding 500 ms to your test suite (5000 ms seemed pretty egregious).

serhalp commented:
Yeah, but like I was saying with the way the test suite is set up it would be tricky. You need the mock server to respond differently on nth attempt based on an unchanging message. I guess it could use a counter in the closure, but that seems a little messy.

vadimdemedes commented:
Thank you!
